### PR TITLE
API to list discovery notebooks from ES backend

### PR DESCRIPTION
**Pull request by cabhishek**
- Load discovery notebooks from elastic index
- New mapping file `commuter.es.mapping.json` for `api/v1/discovery`
- `generateMapping.js` script to create `commuter.es.mapping.json` file
- New config env variable `COMMUTER_ES_HOST` to set elastic host

### CODE REVIEW

File: packages/commuter-server/src/resources/generateMapping.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  }
+});
+
+//IIFE
+{
+  fetch(nbformatV4Schema)
+    .then(res => res.json())
+    .then(getCommuterSchema)
+    .then(writeFile);
+}
```
Comment (rgbkrk) [inline packages/commuter-server/src/resources/generateMapping.js]: I'm assuming this is meant to be run as a utility script? We can also `require` the nbformat package. Let me dig that up.

**Comment by rgbkrk** [comment]
It looks like @minrk has posted https://www.npmjs.com/package/minrk-nbformat-schema while the namespace on the nbformat repo lists `nbformat-schema`. https://github.com/jupyter/nbformat/blob/master/package.json#L2

I'll go ahead and publish nbformat and nbschema with these updates to the format.

File: packages/commuter-server/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        }}
+    ]
+}'
+```
+
+### Delete Index
+
+```
+curl -XDELETE http://<ES_HOST>:7104/commuter
+```
```
Comment (rgbkrk) [inline packages/commuter-server/README.md]: Thanks for the documentation here.

File: packages/commuter-server/src/resources/generateMapping.js (hunk 1 above)
Comment (rgbkrk) [inline packages/commuter-server/src/resources/generateMapping.js]: Alright, you can now `require('nbschema').v4` to get the schema which means you can turn this into:

```
const nbformatv4Schema = require('nbschema').v4
const commuterSchema = getCommuterSchema(nbformatv4Schema);
writeFile(commuterSchema);
```

Which also means we can drop the `node-fetch` dependency.

File: packages/commuter-server/src/resources/nteract.metadata.schema.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+                "description": "Short description about the notebook",
+                "type": "string"
+            },
+            "image": {
+                "description": "Display image url",
+                "type": "string"
+            }
+        }
+    }
+}
```
Comment (rgbkrk) [inline packages/commuter-server/src/resources/nteract.metadata.schema.json]: Cool, we need to move this into nteract/nteract as well. We can probably make a new package there for notebook metadata, cell metadata, and output metadata.

Comment (cabhishek) [inline packages/commuter-server/src/resources/nteract.metadata.schema.json]: yeah, maybe open an issue for tracking?

**Comment by minrk** [comment]
@rgbkrk I forgot to hit publish after the last nbformat release. It's up now.

File: packages/commuter-server/src/resources/generateMapping.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  //   "oneOf": [
+  //   {
+  //     "type": "string"
+  //   },
+  //   {
+  //     "type": "object"
+  //   }
+  // ]
+  properties.language_info.properties.codemirror_mode.type = "object";
+  // ES doesn't support array types. Just use string type
```
Comment (cabhishek) [inline packages/commuter-server/src/resources/generateMapping.js]: @rgbkrk We might have to normalize it before we index the document. And, I couldn't find a way to tell ElasticSearch to do it.

File: packages/commuter-server/src/routes/contents.js [hunk 5]
```diff
@@ -1,8 +1,6 @@
 const express = require("express"),
   router = express.Router(),
-  listObjects = require("./../services/s3").listObjects,
-  getObject = require("./../services/s3").getObject,
-  isDir = require("./util").isDir;
+  { listObjects, getObject, isDir } = require("./../services/s3");
```
Comment (rgbkrk) [inline packages/commuter-server/src/routes/contents.js]: `isDir` is not part of `services/s3`, I'll get this fixed up

Comment (cabhishek) [inline packages/commuter-server/src/routes/contents.js]: ah yes. That's a typo. `isDir` is already imported from `util.js`

Comment (rgbkrk) [inline packages/commuter-server/src/routes/contents.js]: Ok, I'll switch my PR to use util instead.

**Pull request merged**